enhancements/api-review/add-unqualified-search-registries.md

```diff
@@ -0,0 +1,210 @@
+	// +optional
+	UnqualifiedSearchRegistries []string `json:"unqualifiedSearchRegistries,omitempty"`
+}
+```
+
+The containerRuntimeConfig controller in the MCO already watches the cluster-wide images.config.openshift.io CR for the allowed, blocked, and insecure registries. It will now watch for `unqualified-search-registries` as well and update `/etc/containers/registries.conf.d` accordingly.
+
+An example images.config.openshift.io CR will look like:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Image
+metadata:
+  name: cluster
+spec:
+  registrySources:
+    unqualifiedSearchRegistries:
+    - "reg1.io"
+    - "reg2.io"
+    - "reg3.io"
+```
+
+### User Stories
+
+#### As a user, I would like to use image short names when running my workloads
+
+The user can set the `unqualified-search-registries` with a list of registries to check when pulling an image using short names.
+The user can run `oc edit images.config.openshift.io cluster` and add `UnqualifiedSearchRegistries` under `RegistrySources`. Once this is done, the containerRuntimeConfig controller will roll out the changes to the nodes.
+
+#### As a user, I would like to use multiple internal registries to pull my images
+
+The user can use multiple internal registries to pull images with short names without having to change the image spec every time that the registries' DNS changes. This can be done by configuring the list of `unqualified-search-registries` to reflect the changes in the internal registries names by running `oc edit images.config.openshift.io cluster` and add `UnqualifiedSearchRegistries` under `RegistrySources`. Once this is done, the containerRuntimeConfig controller will roll out the changes to the nodes.
+
+### Implementation Details/Notes/Constraints
+
+Implementing this enhancement requires changes in:
+- openshift/api
+- openshift/machine-config-operator
+
+This is what the `/etc/containers/registries.conf` file currently looks like on the nodes:
+
+```
+unqualified-search-registries = ['registry.access.redhat.com', 'docker.io']
+
+[[registry]]
+...
+```
+
+This is an example of the cluster wide images.config.openshift.io:
+
+```
+apiVersion: config.openshift.io/v1
+kind: Image
+metadata:
+  name: cluster
+spec:
+  registrySources:
+    unqualifiedSearchRegistries:
+    - "reg1.io"
+    - "reg2.io"
+    - "reg3.io"
+```
+
+The above Image CR will create a drop-in file at `/etc/containers/registries.conf.d` on each file, which will look like:
+
+```
+unqualified-search-registries = ['reg1.io', 'reg2.io', 'reg3.io']
+```
+
+Note: adding a drop-in file at `/etc/containers/registries.conf.d` completely overrides the default `unqualified-search-registries` list from `/etc/container/registries.conf`. This allows the user to set their list in their own priority order without it ever falling back to the default list.
+
+The new list of `unqualified-search-registries` will be appended to the default list that was already on the nodes.
```

Comment:
This line is confusing after the previous paragraph where we say that the new setting overrides the default. Could you clarify this?

Reply:
Oops, that was a remnant from the previous plan. Fixed now.